File: isma-core/src/main/java/ru/nstu/isma/core/sim/controller/HybridSystemIntgResult.java
package ru.nstu.isma.core.sim.controller;

import ru.nstu.isma.core.sim.controller.gen.EquationIndexProvider;
import ru.nstu.isma.intg.api.IntgMetricData;
import ru.nstu.isma.intg.api.IntgResultPointProvider;

/**
 * @author Maria Nasyrova
 * @since 06.10.2015
 */
public class HybridSystemIntgResult {

    private EquationIndexProvider equationIndexProvider;
    private IntgMetricData metricData;
    private IntgResultPointProvider resultPointProvider;

    public EquationIndexProvider getEquationIndexProvider() {
        return equationIndexProvider;
    }

    public void setEquationIndexProvider(EquationIndexProvider equationIndexProvider) {
        this.equationIndexProvider = equationIndexProvider;
    }

    public IntgMetricData getMetricData() {
        return metricData;
    }

    public void setMetricData(IntgMetricData metricData) {
        this.metricData = metricData;
    }

    public IntgResultPointProvider getResultPointProvider() {
        return resultPointProvider;
    }

    public void setResultPointProvider(IntgResultPointProvider resultPointProvider) {
        this.resultPointProvider = resultPointProvider;
    }
}

File: isma-next-core/src/main/kotlin/ru/nstu/isma/next/core/sim/controller/HybridSystemIntgResult.kt
package ru.nstu.isma.next.core.sim.controller

import ru.nstu.isma.intg.api.IntgMetricData
import ru.nstu.isma.intg.api.IntgResultPointProvider
import ru.nstu.isma.next.core.sim.controller.gen.EquationIndexProvider

/**
 * @author Maria Nasyrova
 * @since 06.10.2015
 */
class HybridSystemIntgResult(
        var equationIndexProvider: EquationIndexProvider? = null,
        var metricData: IntgMetricData? = null,
        var resultPointProvider: IntgResultPointProvider? = null)